src/components/organisms/WithNavigationBar/WithNavigationBar.tsx

```diff
@@ -16,17 +31,71 @@ export const WithNavigationBar: React.FunctionComponent<PropsType> = ({
   redirectionUrl,
   fullscreen,
   children,
-  hasBackButton,
-}) => (
-  <>
-    {/* @debt remove backButton from Navbar */}
-    <NavBar redirectionUrl={redirectionUrl} hasBackButton={hasBackButton} />
-    <div className={`navbar-margin ${fullscreen ? "fullscreen" : ""}`}>
-      {children}
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+  const venue = useSelector(currentVenueSelectorData);
+  const navigateToHomepage = useCallback(() => {
+    const venueLink =
+      redirectionUrl ?? venueId ? venueInsideUrl(venueId ?? "") : "/";
+
+    window.location.href = venueLink;
+  }, [redirectionUrl, venueId]);
+
+  const leftComponent = (
+    <div className="nav-logos">
+      <div className="nav-sparkle-logo">
+        <div className="nav-sparkle-logo_small" onClick={navigateToHomepage} />
+      </div>
+      {venue && (
+        <>
+          <NavbarSchedule />
+          <VenuePartygoers />
+        </>
+      )}
     </div>
-    <Footer />
-  </>
-);
+  );
+
+  const rightComponent = user ? (
+    <div className="navbar-links">
+      {venue && <NavSearchBar />}
+
+      {venue?.showGiftATicket && (
+        <OverlayTrigger
+          trigger="click"
+          placement="bottom-end"
+          rootClose={true}
+          overlay={
+            <Popover id="gift-popover">
+              <Popover.Content>
+                <GiftTicketModal />
+              </Popover.Content>
+            </Popover>
+          }
+        >
+          <span className="private-chat-icon">
+            <div className="navbar-link-gift" />
+          </span>
+        </OverlayTrigger>
+      )}
```

Comment:
I agree, but there is 99% chance that this component is going to be removed because it was used for Burning man and never again ever since. Added a debt in the GiftTicketModal for it.